src/test/java/no/sikt/nva/pubchannels/handler/FetchJournalByIdentifierAndYearHandlerTest.java

```diff
@@ -0,0 +1,188 @@
+                        ))
+                        .build();
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+
+        var problem = response.getBodyObject(Problem.class);
+
+        assertThat(problem.getDetail(),
+                   is(equalTo("Invalid path parameter (identifier). Must be a UUID version 4.")));
+    }
+
+    private String randomYear() {
+        return Integer.toString(YEAR_START + randomInteger(RANDOM_YEAR_BOUND));
+    }
+
+    @Test
+    public void shouldReturnBadGatewayWhenChannelRegistryIsUnavailable() throws IOException {
+        var httpClient = WiremockHttpClient.create();
+        var dataportenBaseUri = URI.create("https://localhost:9898");
+        var apiDomain = "localhost";
+        var basePath = "publication-channels";
+        var publicationChannelSource = new DataportenPublicationChannelSource(httpClient,
+                                                                              dataportenBaseUri,
+                                                                              apiDomain,
+                                                                              basePath);
+        this.handlerUnderTest = new FetchJournalByIdentifierAndYearHandler(publicationChannelSource);
+
+        var input = new HandlerRequestBuilder<Void>(dtoObjectMapper)
+                        .withPathParameters(Map.of(
+                            "identifier", UUID.randomUUID().toString(),
+                            "year", randomYear()
+                        ))
+                        .build();
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_GATEWAY)));
+
+        var problem = response.getBodyObject(Problem.class);
+
+        assertThat(problem.getDetail(),
+                   is(equalTo("Unable to reach upstream!")));
+    }
+
+    @Test
+    public void shouldReturnBadGatewayWhenChannelRegistryReturnsNonSuccess() throws IOException {
+        var identifier = UUID.randomUUID().toString();
+        var year = randomYear();
+
+        mockRegistry.notFoundJournal(identifier, year);
+
+        var input = new HandlerRequestBuilder<Void>(dtoObjectMapper)
+                        .withPathParameters(Map.of(
+                            "identifier", identifier,
+                            "year", year
+                        ))
+                        .build();
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_GATEWAY)));
```

Comment:
BadGateway on 404?

Reply:
Oops, how did I miss that :) I am fixing this one.

Reply:
Fixed